workload/jobs.py

```diff
@@ -886,8 +886,42 @@ def _generate_task_graph(
         task_graph_deadline = release_time + weighted_task_graph_length.fuzz(
             deadline_variance, deadline_bounds
         )
-        for task in task_graph.get_nodes():
-            task.update_deadline(task_graph_deadline)
+        if _flags and _flags.enable_deadline_decomposition:
+            stages_info = {}
+            stages = set([])
+            for task in task_graph.topological_sort():
+                stage = 0
+                for previous_task in task_graph.get_parents(task):
+                    stage = max(stage, stages_info.get(previous_task, 0) + 1)
+                stages_info[task] = stage
+                stages.add(stage)
+
+            critical_path = task_graph.get_longest_path(
+                weights=lambda task: (task.slowest_execution_strategy.runtime.time)
+            )
+            critical_path_time = sum(
+                [t.slowest_execution_strategy.runtime.time for t in critical_path]
+            )
```

Comment:
```suggestion
            critical_path_time = sum(
                [t.slowest_execution_strategy.runtime for t in critical_path],
                start=EventTime.zero(),
            )
```

I'm expecting a lot of usage from Spark to be in `EventTime.Unit.S` so I want to minimize anything that assumes the same time units.